restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -84,6 +91,90 @@ static Object translateCanonicalKeyToAlternativeKeyIfNeeded(Object canonicalKey,
     }
   }
 
+  /**
+   *
+   * @param dataSchema
+   * @param dataMap
+   * @return a new different data map that contains original data plus the default values
+   */
+  public static DataMap fillInDefaultValues(DataSchema dataSchema, DataMap dataMap)
+  {
+    DataMap dataDefaultFilled = new DataMap();
+    if (dataMap != null)
+    {
+      dataDefaultFilled.putAll(dataMap);
+    }
+
+    if (dataSchema instanceof TyperefDataSchema)
+    {
+      TyperefDataSchema typerefDataSchema = (TyperefDataSchema) dataSchema;
+      return fillInDefaultValues(typerefDataSchema.getDereferencedDataSchema(), dataDefaultFilled);
+    }
+    else if (dataSchema instanceof RecordDataSchema)
+    {
+      RecordDataSchema recordDataSchema = (RecordDataSchema) dataSchema;
+      for (RecordDataSchema.Field field : recordDataSchema.getFields())
+      {
+        DataSchema fieldSchema = field.getType();
+        if (fieldSchema instanceof RecordDataSchema)
+        {
+          if (dataDefaultFilled.containsKey(field.getName()))
+          {
+            DataMap fieldDefault = (DataMap) field.getDefault();
+            DataMap fieldDataAssigned = (DataMap) dataDefaultFilled.get(field.getName());
+            if (fieldDataAssigned != null)
```

Comment:
redundant as the containsKey check above ensures the value wouldn't be null

Reply:
yes